### PR TITLE
reduced chances of default ad being called

### PR DESCRIPTION
**Pull request by sgreene570**
Everything compiled, hopefully it doesn't break everything. No promises!

### CODE REVIEW

File: plug.go [hunk 1]
```diff
@@ -10,7 +10,41 @@ type Plug struct {
 	ViewsRemaining int
 }
 
+func (p Plug) isDefault() bool {
+	if p.ViewsRemaining >= 0 {
```
Comment (liam-middlebrook) [inline plug.go]: nit: pls condense to single line body

File: plug.go [hunk 2]
```diff
@@ -10,7 +10,41 @@ type Plug struct {
...
+	for i := 0; i < len(plugs); i++ {
+		if plug.isDefault() {
+			numAds++
+		}
+	}
+
+	var defaults []Plug
+	defaults = make([]Plug, numAds)
+	defaultPos := 0
+	for i := 0; i < len(plugs); i++ {
```
Comment (liam-middlebrook) [inline plug.go]: Can't this be compacted to a single for loop?

**Review by liam-middlebrook — CHANGES_REQUESTED** [review]
From a quick read I don't believe that this will decrease the amount of default ads that are shown. It looks like it might actually increase the number because line 47 as well as line 49 could choose a default ad.

**Comment by sgreene570** [comment]
Totally missed the contradiction in line 49. I will add link lists to clean things up (I needed to declare arrays with fixed sizes, that's why I had 2 for loops).

File: plug.go [hunk 3]
```diff
@@ -10,7 +10,27 @@ type Plug struct {
 	ViewsRemaining int
 }
 
+func (p Plug) isDefault() bool {
```
Comment (liam-middlebrook) [inline plug.go]: nit: pls capitalize the leading letter of a function

File: plug.go [hunk 4]
```diff
@@ -10,7 +10,27 @@ type Plug struct {
...
+	for i := 0; i < len(plugs); i++ {
+		if plugs[i].isDefault() {
+			defaults = append(defaults, plugs[i])
+		} else {
+			customs = append(customs, plugs[i])
+		}
+	}
+	// Decide whether to chose default ad or user submitted ad
+	var pickDefault int = rand.Intn(100)
+	if pickDefault >= 85 && len(defaults) != 0{
```
Comment (liam-middlebrook) [inline plug.go]: nit: pls move the constant 85 to a `const int` at the top of the file

**Review by liam-middlebrook — CHANGES_REQUESTED** [review]
fix the nits, otherwise lgtm!

**Comment by sgreene570** [comment]
Fixed the nits.

**Review by liam-middlebrook — APPROVED** [review]
lgtm

**Pull request merged**